mindmeld/augmentation.py

```diff
@@ -0,0 +1,315 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Cisco Systems, Inc. and others.  All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains the data augmentation processes for MindMeld."""
+
+import logging
+import re
+
+from abc import ABC, abstractmethod
+from tqdm import tqdm
+
+from .components._util import _is_module_available, _get_module_or_attr
+from .constants import get_pattern
+from .models.helpers import register_augmentor
+
+logger = logging.getLogger(__name__)
+
+# pylint: disable=R0201
+
+SUPPORTED_LANG_CODES = ['en', 'es', 'fr', 'pt', 'it']
+
+
+class UnsupportedLanguageError(Exception):
+    pass
+
+
+class Augmentor(ABC):
+    """
+    Abstract Augmentor class.
+    """
+
+    def __init__(self, lang, paths, path_suffix, resource_loader):
+        """Initializes an augmentor.
+
+        Args:
+            lang (str): The lang code for paraphrasing
+            paths (list): Path rules for fetching relevant files to Paraphrase.
+            path_suffix (str): Suffix to be added to new augmented files.
+            resource_loader (object): Resource Loader object for the application.
+        """
+        self.lang = lang
+        self.paths = paths
+        self.path_suffix = path_suffix
+        self._resource_loader = resource_loader
+        self._check_dependencies()
+        self._check_lang_support()
+
+    def _check_dependencies(self):
+        if not _is_module_available("torch"):
+            raise ModuleNotFoundError(
+                "Library not found: 'torch'. Run 'pip install mindmeld[augment]' to install."
+            )
+
+        if not _is_module_available("transformers"):
+            raise ModuleNotFoundError(
+                "Library not found: 'transformers'. Run 'pip install mindmeld[augment]' to install."
+            )
+
+    def _check_lang_support(self):
+        if self.lang not in SUPPORTED_LANG_CODES:
+            raise UnsupportedLanguageError(
+                f"'{self.lang}' is not supported yet. "
+                "English, Spanish, French, Portuguese and Italian are currently supported."
```

Comment:
Nit: Add language codes as well. 

"English(en), Spanish(es), French(fr), Portuguese(pt) and Italian(it) are currently supported."

Reply:
Also add Romanian (ro)